Ecal/src/Ecal/EcalClusterProducer.cxx

```diff
@@ -14,57 +14,68 @@ EcalClusterProducer::EcalClusterProducer(const std::string& name,
-  algo_coll_name_ = parameters.get<std::string>("algo_coll_name");
-  algo_name_ = parameters.get<std::string>("algo_name");
-  cluster_coll_name_ = parameters.get<std::string>("cluster_coll_name");
-  clue_ = parameters.get<bool>("CLUE");
-  nbr_of_layers_ = parameters.get<int>("nbr_of_layers");
-  reclustering_ = parameters.get<bool>("reclustering");
+void EcalClusterProducer::configure(framework::config::Parameters& ps) {
+  cutoff_ = ps.get<double>("cutoff");
+  seed_threshold_ = ps.get<double>("seed_threshold");
+
+  dc_ = ps.get<double>("dc");
+  rhoc_ = ps.get<double>("rhoc");
+  deltac_ = ps.get<double>("deltac");
+  deltao_ = ps.get<double>("deltao");
+
+  rec_hit_coll_name_ = ps.get<std::string>("rec_hit_coll_name");
+  rec_hit_pass_name_ = ps.get<std::string>("rec_hit_pass_name");
+  algo_coll_name_ = ps.get<std::string>("algo_coll_name");
+  algo_name_ = ps.get<std::string>("algo_name");
+  cluster_coll_name_ = ps.get<std::string>("cluster_coll_name");
+  clue_ = ps.get<bool>("CLUE");
+  nbr_of_layers_ = ps.get<int>("nbr_of_layers");
+  reclustering_ = ps.get<bool>("reclustering");
 }
 
 void EcalClusterProducer::produce(framework::Event& event) {
   const auto& ecal_hits{event.getCollection<ldmx::EcalHit>(rec_hit_coll_name_,
                                                            rec_hit_pass_name_)};
   if (ecal_hits.size() == 0) {
-    // don't do anything if there are no ECal hits_
+    // don't do anything if there are no ECal hits
+    ldmx_log(fatal) << "No ECal hits found... exiting";
     return;
   }
 
   if (clue_) {
+    ldmx_log(info) << "Using CLUE clustering algorithm";
     CLUE cf;
     cf.cluster(ecal_hits, dc_, rhoc_, deltac_, deltao_, nbr_of_layers_,
                reclustering_);
+    ldmx_log(debug) << "CLUE algorithm finished clustering";
     std::vector<IntermediateCluster> wc_vec = cf.getClusters();
     std::vector<IntermediateCluster> f_wc_vec = cf.getFirstLayerCentroids();
+    ldmx_log(debug) << "Got " << wc_vec.size() << " clusters with "
+                    << f_wc_vec.size() << " first layer centroids";
 
     auto n_loops = cf.getNLoops();
-    histograms_.fill("nLoops", n_loops);
-    histograms_.fill("nClusters", wc_vec.size());
-    if (reclustering_)
+    ldmx_log(debug) << "Number of clustering loops: " << n_loops;
+
+    if (reclustering_) {
+      ldmx_log(debug) << "Reclustererd initial number of clusters: "
+                      << cf.getInitialClusterNbr()
+                      << ", final number of clusters: " << wc_vec.size();
       histograms_.fill("recluster", cf.getInitialClusterNbr(), wc_vec.size());
+    }
 
     std::vector<ldmx::EcalCluster> ecal_clusters;
+    ldmx_log(debug) << "Filling " << wc_vec.size()
+                    << " clusters into ecal_clusters";
     for (int a_wc = 0; a_wc < wc_vec.size(); a_wc++) {
       ldmx::EcalCluster cluster;
 
       cluster.setEnergy(wc_vec[a_wc].centroid().E());
-      cluster.setCentroidXYZ(wc_vec[a_wc].centroid().Px(),
```

Comment:
I dont understand why this was on the momentum in the first place, that's not something we can measure with the ECal